File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        result = [0] * (n+1)
        offset = 1
        for i in range(1,n+1):
            if i == offset * 2:
                offset = offset * 2
            result[i] = result[i-offset] + 1
        return result

File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] result = new int[n+1];
        result[0] = 0;
        int offset = 1;
        for(int i=1;i<n+1;i++){
            if(i == offset * 2){
                offset = offset * 2;
            }
            result[i] = result[i-offset] + 1;
        }
        return result;
    }
}